lib/binding_web/binding.js

```diff
@@ -842,78 +842,90 @@ class Language {
           const operator = steps[0].value;
           let isPositive = true;
           let matchAll = true;
+          let captureName;
           switch (operator) {
             case 'any-not-eq?':
-              isPositive = false;
-              matchAll = false;
-            case 'any-eq?':
-              matchAll = false;
             case 'not-eq?':
               isPositive = false;
+            case 'any-eq?':
             case 'eq?':
               if (steps.length !== 3) throw new Error(
-                `Wrong number of arguments to \`#eq?\` predicate. Expected 2, got ${steps.length - 1}`
+                `Wrong number of arguments to \`#${operator}\` predicate. Expected 2, got ${steps.length - 1}`
               );
               if (steps[1].type !== 'capture') throw new Error(
-                `First argument of \`#eq?\` predicate must be a capture. Got "${steps[1].value}"`
+                `First argument of \`#${operator}\` predicate must be a capture. Got "${steps[1].value}"`
               );
+              let eqMatchAll = !operator.startsWith('any-');
+              let eqIsPositive = isPositive;
```

Comment:
You may wonder what the point of this is. It makes sense in my head, but I haven't quite come up with the test case that would reveal why this is necessary:

`isPositive` is a variable whose value can change each time through this loop. But when we add to the `textPredicates` list, we're defining a closure that will consult `isPositive` at some time in the future — possibly after it has changed.

If we instead define a variable with `let` or `const` within an inner block, we'll get what we want: a snapshot of`isPositive`'s value at the time of evaluation.

I'm pretty sure I've still gotten this wrong, though, because a `case` consequence clause doesn't create its own implicit block. Bear with me.

Hard to think of cases that would illustrate this problem without being nonsensical, but this might be one:

```scm
(
  (comment) @foo
  (#match? @foo "^//")
  (#not-match? @foo "^///")
)
```

Reply:
Never mind — I think this can't happen because each new predicate has its own block scope from the `for` loop. I'll simplify this and update the PR.